### PR TITLE
[metrics] Document view level cardinality limit

### PR DESCRIPTION
**Pull request by reyang**
Follow up #5312.

### CODE REVIEW

File: docs/metrics/getting-started-console/README.md [hunk 1]
```diff
@@ -69,15 +69,16 @@ MyFruitCounter.Add(2, new("name", "lemon"), new("color", "yellow"));
...
+An OpenTelemetry [MeterProvider](#meterprovider) is configured to subscribe to
+an instrument named "MyFruitCounter" from the Meter
+`MyCompany.MyProduct.MyLibrary`, and aggregate the measurements in-memory with a
+[cardinality limit](../README.md#cardinality-limits) of `10`. The pre-aggregated
+metrics are exported to a `ConsoleExporter`.
 
 ```csharp
 var meterProvider = Sdk.CreateMeterProviderBuilder()
     .AddMeter("MyCompany.MyProduct.MyLibrary")
+    .AddView(instrumentName: "MyFruitCounter", new MetricStreamConfiguration { CardinalityLimit = 10 })
```
Comment (cijothomas) [inline docs/metrics/getting-started-console/README.md:81]: this is not required in the 'getting-started' doc.

**Review by rajkumar-rangaraj — APPROVED** [review]

File: docs/metrics/getting-started-console/README.md (hunk 1 above)
Comment (reyang) [reply docs/metrics/getting-started-console/README.md]: Tweaked a little bit.

File: docs/metrics/customizing-the-sdk/README.md [hunk 2]
```diff
@@ -444,10 +444,16 @@ AnotherFruitCounter.Add(5, new("name", "banana"), new("color", "yellow")); // Ex
 AnotherFruitCounter.Add(4, new("name", "mango"), new("color", "yellow")); // Not exported
 ```
 
-> [!NOTE]
-> The above limit is *per* metric stream, and applies to all the metric
-streams. There is no ability to apply different limits for each instrument at
-this moment.
+To set the [cardinality limit](../README.md#cardinality-limits) at view level,
```
Comment (cijothomas) [inline docs/metrics/customizing-the-sdk/README.md:447]: at view level -> per individual metric -- might be more beginner friendly :)

File: docs/metrics/getting-started-console/README.md (hunk 1 above)
Comment (rajkumar-rangaraj) [reply docs/metrics/getting-started-console/README.md]: Looks good now.

File: docs/metrics/getting-started-console/README.md [hunk 3]
```diff
@@ -82,6 +82,20 @@ var meterProvider = Sdk.CreateMeterProviderBuilder()
     .Build();
 ```
 
+> [!NOTE]
+> If you need to collect metrics with cardinality higher than the default limit
+  `2000`, please follow the [cardinality
+  limits](../README.md#cardinality-limits) guidance. Here is a quick example of
+  how to change the cardinality limit to `10`:
```
Comment (cijothomas) [inline docs/metrics/getting-started-console/README.md:89]:
```suggestion
  how to change the cardinality limit to `10` for this particular metric:
```

File: docs/metrics/customizing-the-sdk/README.md (hunk 2 above)
Comment (reyang) [reply docs/metrics/customizing-the-sdk/README.md]: Updated.

**Review by cijothomas — APPROVED** [review]

**Review by vishweshbankwar — APPROVED** [review]

**Review by Yun-Ting — APPROVED** [review]

File: docs/metrics/README.md [hunk 4]
```diff
@@ -356,18 +356,13 @@ predictable and reliable behavior when excessive cardinality happens, whether it
 was due to a malicious attack or developer making mistakes while writing code.
 
 OpenTelemetry has a default cardinality limit of `2000` per metric. This limit
-can be configured at `MeterProvider` level using
+can be configured at `MeterProvider` level using the
 `SetMaxMetricPointsPerMetricStream` method, or at individual
 [view](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#view)
-level. Refer to this
+level using `MetricStreamConfiguration.CardinalityLimit`. Refer to this
 [doc](../../docs/metrics/customizing-the-sdk/README.md#changing-maximum-metricpoints-per-metricstream)
```
Comment (Yun-Ting) [inline docs/metrics/README.md:363]: This section can be updated to be "changing-the-cardinalityLimit-of-a-metric" given the name standardization effort: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5328/files.

File: docs/metrics/README.md [hunk 5]
```diff
@@ -356,18 +356,13 @@ predictable and reliable behavior when excessive cardinality happens, whether it
 was due to a malicious attack or developer making mistakes while writing code.
 
 OpenTelemetry has a default cardinality limit of `2000` per metric. This limit
-can be configured at `MeterProvider` level using
+can be configured at `MeterProvider` level using the
 `SetMaxMetricPointsPerMetricStream` method, or at individual
```
Comment (Yun-Ting) [inline docs/metrics/README.md:360]:
```suggestion
`SetCardinalityLimit` method, or at individual
```

**Comment by Yun-Ting** [comment]
Might be nice to _explicitly_ point out in the metrics readme file that cardinality limit will be pre-allocated to prevent user from abusing it.

**Comment by reyang** [comment]
> Might be nice to _explicitly_ point out in the metrics readme file that cardinality limit will be pre-allocated to prevent user from abusing it.

This is captured here https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/metrics#memory-preallocation.

**Comment by CodeBlanch** [comment]
I merged this and then put some updates on #5328

**Pull request merged**